lib/spack/spack/solver/asp.py

```diff
@@ -2478,6 +2478,23 @@ def define_runtime_constraints(self):
             if hasattr(compiler_cls, "runtime_constraints"):
                 compiler_cls.runtime_constraints(spec=compiler.spec, pkg=recorder)
 
+            # Inject libc from available compilers, on Linux
+            if not compiler.available:
+                continue
+
+            if spack.platforms.host().name == "linux":
```

Comment:
We may also include BSD (later?). Plus at some point this should be a target not host condition :thinking:

Reply:
Would you be fine with https://github.com/spack/spack/pull/43190/commits/85d8bb10705c613c5e3ca1ffbcde3bbc74c18488 for now (which just adds a function where to encode when libc compatibility is used)?